apps/dashboard/src/lib/components/Course/components/Lesson/Exercise/ViewMode.svelte

```diff
@@ -80,7 +91,89 @@
     });
   };
 
-  function onSubmit(id, value, moveToNextQuestion = false) {
+  const { input, handleSubmit, completion } = useCompletion({
+    api: '/api/completion/gradingprompt',
+    onFinish: async () => {
+      try {
+        const responseData = $completion.replace('```json', '').replace('```', '');
+
+        let aiResponses = [];
+        try {
+          // Parse the modified response data as JSON
+          aiResponses = JSON.parse(responseData);
+        } catch (error) {
+          console.error('Error parsing AI response', error);
+        }
+
+        $questionnaire.questions.forEach((question) => {
+          const { id, points, question_type_id } = question;
+          let questionAnswer = submissionResponse?.res?.data.find(
+            (answer) => answer.question_id === id
+          );
+
+          console.log('this is the question answer', questionAnswer);
+          if (question_type_id !== QUESTION_TYPE.TEXTAREA) {
+            const answer = $questionnaireMetaData.answers[question.name];
+            let score = points / answer.length;
+            $questionnaireMetaData.grades = {
+              ...$questionnaireMetaData.grades,
```

Comment:
yea you did but i suggested that we use the AI logic for auto grading and you agreed...i don't think it is right to give out the maximum score for paragraph question regardless of the answer provided, it only gives a false positive to student taking a self paced course

Reply:
Yeah I get that but it can add more overhead on our AI functionality if we have to grade all paragraph questions from a cost perspective

Besides self paced courses are for students to take the course how they see fit. No one is verifying if they got the right answer or not. 

In addition there should be more radio or checkbox questions because the teacher isn't grading.

My point here is that the score is less important for self paced courses cause the idea is that the creator doesn't want to monitor the student

Reply:
@Chifez Please let's do this first on Monday